### PR TITLE
[rtl872x] fix wifi/ble coexistence issues; USB re-enum workarounds

### PR DESCRIPTION
**Pull request by avtolstoy**
TODO

### CODE REVIEW

File: hal/src/rtl872x/ble_hal.cpp [hunk 1]
```diff
@@ -229,7 +238,7 @@ class BleEventDispatcher {
     void* allEvtQueue_;
     void* ioEvtQueue_;
     bool started_;
-    static constexpr uint8_t BLE_EVENT_THREAD_PRIORITY = OS_THREAD_PRIORITY_NETWORK;
+    static constexpr uint8_t BLE_EVENT_THREAD_PRIORITY = OS_THREAD_PRIORITY_NETWORK - 1; // IMPORTANT: below network for coexistence to work!
```
Comment (eberseth) [inline hal/src/rtl872x/ble_hal.cpp:241]: Any thread managing WiFi would effectively be higher priority than the BLE thread?

File: hal/src/rtl872x/ble_hal.cpp [hunk 2]
```diff
@@ -1632,22 +1653,21 @@ int BleGap::startScanning(hal_ble_on_scan_result_cb_t callback, void* context) {
...
             for (int i = 0; i < LE_SCAN_STOP_RETRIES; i++) {
                 // This has seen failing a number of times at least
                 // with btgap logging enabled. Retry a few times just in case,
                 // otherwise the next scan operation fails with invalid state.
-                auto r = le_scan_stop();
+                auto r = WRAP_BLE_EVT_LOCK(le_scan_stop());
                 if (r == GAP_CAUSE_SUCCESS) {
-                    LOCAL_DEBUG("wait GAP_SCAN_STATE_IDLE");
-                    if (!waitState(BleGapDevState().scan(GAP_SCAN_STATE_IDLE))) {
+                    // IMPORTANT: have to poll here, for some reason we may not get notified
```
Comment (eberseth) [inline hal/src/rtl872x/ble_hal.cpp:1664]: Does this have anything to do with missing events/interrupts from the SDK?

**Review by eberseth — APPROVED** [review]

**Comment by Kategrode** [comment]
https://app.shortcut.com/particle/story/125195/device-os-5-7-0-breaks-serial-port-access-on-msom

**Comment by Kategrode** [comment]
https://app.shortcut.com/particle/story/125856/wifi-scan-failure-for-jacuzzi

**Pull request merged**